maxfw/tests/test_image_utils.py

```diff
@@ -0,0 +1,171 @@
+Image.open('maxfw/tests/test_image.jpg').convert('RGBA').save(stream, 'PNG')
+test_input = stream.getvalue()
+
+
+def test_imageprocessor_read():
+    """Test the Imageprocessor."""
+
+    # Test with 4 channels
+    transform_sequence = [ToPILImage('RGBA')]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (678, 1024, 4)
+
+    # Test with 3 channels
+    transform_sequence = [ToPILImage('RGB')]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (678, 1024, 3)
+
+
+def test_imageprocessor_resize():
+    """Test the Imageprocessor's resize function."""
+
+    # Resize to 200x200
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(200, 200))]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 4)
+
+    # Resize to 2000x2000
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(2000, 2000))]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (2000, 2000, 4)
+
+
+def test_imageprocessor_grayscale():
+    """Test the Imageprocessor's grayscale function."""
+
+    # Using the standard 1 output channel
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(200, 200)), Grayscale()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200)
+
+    # Using 3 output channels
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(200, 200)), Grayscale(num_output_channels=3)]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 3)
+
+    # Using 4 output channels
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(200, 200)), Grayscale(num_output_channels=4)]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 4)
+
+
+def test_imageprocessor_normalize():
+    """Test the Imageprocessor's normalize function."""
+
+    # Test normalize
+    transform_sequence = [ToPILImage('RGBA'), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+    # Test normalize
+    transform_sequence = [ToPILImage('RGB'), Normalize(), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+    # Test normalize
+    transform_sequence = [ToPILImage('L'), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+
+def test_imageprocessor_standardize():
```

Comment:
Here when expanding the `standardize` function we will need to test channel-wise operations.